lib/Dialect/ESI/AppID.cpp

```diff
@@ -198,11 +204,13 @@ FailureOr<ArrayAttr> AppIDIndex::getAppIDPathAttr(hw::HWModuleLike fromMod,
     if (getAppID(*op))
       break;
 
-    if (auto inst = dyn_cast<hw::HWInstanceLike>(op->getOperation()))
-      fromMod = cast<hw::HWModuleLike>(
-          symCache.getDefinition(inst.getReferencedModuleNameAttr()));
-    else
+    if (auto inst = dyn_cast<hw::HWInstanceLike>(op->getOperation())) {
+      auto moduleNames = inst.getReferencedModuleNamesAttr();
+      assert(moduleNames.size() == 1 && "expected a single reference");
+      fromMod = cast<hw::HWModuleLike>(symCache.getDefinition(moduleNames[0]));
+    } else {
       assert(false && "Search bottomed out");
+    }
```

Comment:
This particular method supports returning an error, which is better than crashing or silently doing the wrong thing. I'm not sure how I want to deal with `instance_choice` here (as I'm not familiar enough with that op). So return an op error with a message like "AppID hierarchies do not yet support references to multiple modules".

Reply:
I don't think you want to see the multi-target version yet at all, so I think an assertion is better suited at this point.

Reply:
Assertions are bad because they don't trigger in release builds. I don't think you want silent continuation and then assumedly a later blow up?

Reply:
Yes, since this can be encountered on valid IR, an error is far more appropriate. The existing `assert`s are actually internal errors which should never be encountered.

Reply:
Since this IR is not generated from FIRRTL, it is a sign of an internal error to see such an op. But I'll convert to explicit errors.